specs/opcode/94MCOPY.md

```diff
@@ -0,0 +1,40 @@
+# MCOPY opcode
+
+## Procedure
+
+The `MCOPY` opcode pops `dest_offset`, `offset` and `size` from the stack.
+It then copies `size` bytes in the current environment from memory at an `offset` to the memory at a `dest_offset`. For out-of-bounds scenarios - the memory is extended with respective gas cost applied.
+
+The gas cost of `MCOPY` opcode consists of two parts:
+
+1. A constant gas cost: `3 gas`
+2. A dynamic gas cost: cost of memory expansion and copying (variable depending on the `size` copied to memory)
+
+## Constraints
+
+1. OpcodeId check
+
+- opId == OpcodeId(0x5E)
+
+2. State Transition
+
+- rw_counter += 3 stack_reads + ???
```

Comment:
Not sure here, need help ...

Reply:
Except for the stack reads I do not see other RW since you'd use the copy table afterwards. This is confirmed in Scroll WIP implementation

`//rw_counter: Transition::Delta(3.expr()),`